torchao/quantization/qat/linear.py

```diff
@@ -219,7 +219,10 @@ def _convert_qat_linear_8da4w(self, module: torch.nn.Module):
                 n_bit = 4
                 (qmin, qmax) = _get_qmin_qmax(n_bit)
                 (s, zp) = get_group_qparams_symmetric(
-                    child.weight, n_bit, config.group_size
+                    child.weight,
+                    n_bit,
+                    config.group_size,
+                    precision=config.scale_precision,
```

Comment:
same here, probably need to set `zp = zp.to(config.zero_point_precision)` to be more correct